.teamcity/src/main/kotlin/configurations/FlakyTestQuarantine.kt

```diff
@@ -0,0 +1,47 @@
+package configurations
+
+import common.BuildToolBuildJvm
+import common.Os
+import common.applyDefaultSettings
+import common.buildToolGradleParameters
+import common.checkCleanM2AndAndroidUserHome
+import common.functionalTestExtraParameters
+import common.functionalTestParameters
+import common.gradleWrapper
+import common.killProcessStep
+import jetbrains.buildServer.configs.kotlin.v2019_2.BuildStep
+import model.CIBuildModel
+import model.Stage
+
+class FlakyTestQuarantine(model: CIBuildModel, stage: Stage, os: Os) : BaseGradleBuildType(stage = stage, init = {
+    id("${model.projectId}_FlakyQuarantine_${os.name.capitalize()}")
+    name = "Flaky Test Quarantine - ${os.name.capitalize()}"
+    description = "Run all flaky tests skipped multiple times"
+
+    applyDefaultSettings(os, BuildToolBuildJvm, 60)
+
+    val testsWithOs = model.stages.flatMap { it.functionalTests }.filter { it.os == os }
```

Comment:
I don't think we should do all stages here, probably only the stages before ready for release, right?